File: C++/Busca/buscabinária.cpp
#include <iostream>
#include <vector>

int busca_binaria(std::vector<int> lista, int alvo) {
    int primeiro = 0;
    int último = lista.size() - 1;
    while (primeiro <= último) {
        int meio = (primeiro + último) / 2;
        if (lista[meio] == alvo) {
            return meio;
        } else if (alvo < lista[meio]) {
            último = meio - 1;
        } else {
            primeiro = meio + 1;
        }
    }
    return -1;
}

int main() {
    std::vector<int> lista = {1, 3, 5, 7, 9, 11};
    int alvo = 5;
    std::cout << busca_binaria(lista, alvo) << std::endl; 
    alvo = 4;
    std::cout << busca_binaria(lista, alvo) << std::endl;
    alvo = 9;
    std::cout << busca_binaria(lista, alvo) << std::endl;
    return 0;
}

File: Python/Busca/buscabinária.py
def busca_binaria(lista, alvo):
    primeiro = 0
    último = len(lista) - 1
    while primeiro <= último:
        meio = (primeiro + último) // 2
        if lista[meio] == alvo:
            return meio
        else: 
            if alvo < lista[meio]:
                último = meio - 1
            else: 
                primeiro = meio + 1
    return False

lista = [1, 3, 5, 7, 9, 11]
alvo = 5
print(busca_binaria(lista, alvo)) 
alvo = 4
print(busca_binaria(lista, alvo)) 
alvo = 9
print(busca_binaria(lista, alvo))